examples_tests/02.ComputeShader/main.cpp

```diff
@@ -1,152 +1,183 @@
 
 	core::smart_refctd_ptr<video::IGPUComputePipeline> compPipeline;
+	core::smart_refctd_ptr<video::IGPUPipelineLayout> layout;
 	{
-		core::smart_refctd_ptr<video::IGPUPipelineLayout> layout;
 		{
 			asset::SPushConstantRange range;
 			range.offset = 0u;
 			range.size = sizeof(uint32_t) * 2u;
 			range.stageFlags = asset::ISpecializedShader::ESS_COMPUTE;
-			layout = driver->createGPUPipelineLayout(&range, &range + 1, core::smart_refctd_ptr_dynamic_cast<video::IGPUDescriptorSetLayout>(am->findGPUObject(ds0layout.get())), nullptr, nullptr, nullptr);
+			layout = device->createGPUPipelineLayout(&range, &range + 1, std::move(ds0layout));
 		}
 		core::smart_refctd_ptr<video::IGPUSpecializedShader> shader;
 		{
-			auto f = core::smart_refctd_ptr<io::IReadFile>(filesystem->createAndOpenFile("../compute.comp"));
-
-			//auto cs_unspec = am->getGLSLCompiler()->createSPIRVFromGLSL(f.get(), asset::ISpecializedShader::ESS_COMPUTE, "main", "comp");
-			asset::IAssetLoader::SAssetLoadParams lp;
-			auto cs_bundle = am->getAsset("../compute.comp",lp);
-			auto cs = core::smart_refctd_ptr_static_cast<asset::ICPUSpecializedShader>(*cs_bundle.getContents().begin());
+			//TODO: Load from "../compute.comp" instead of getting source from src
+			auto cs_unspec = device->createGPUShader(core::make_smart_refctd_ptr<asset::ICPUShader>(src));
+			asset::ISpecializedShader::SInfo csinfo(nullptr, nullptr, "main", asset::ISpecializedShader::ESS_COMPUTE, "cs");
+			auto cs = device->createGPUSpecializedShader(cs_unspec.get(), csinfo);
 
-			auto cs_rawptr = cs.get();
-			shader = driver->getGPUObjectsFromAssets(&cs_rawptr, &cs_rawptr+1)->front();
+			compPipeline = device->createGPUComputePipeline(nullptr, core::smart_refctd_ptr(layout), std::move(cs));
 		}
-
-		compPipeline = driver->createGPUComputePipeline(nullptr, std::move(layout), std::move(shader));
 	}
 
-	auto renderObject = driver->getGPUObjectsFromAssets(&outImgViewRawPtr, &outImgViewRawPtr + 1)->front();
 
-	auto blitFBO = driver->addFrameBuffer();
-	blitFBO->attach(video::EFAP_COLOR_ATTACHMENT0, core::smart_refctd_ptr_dynamic_cast<video::IGPUImageView>(renderObject));
-
-	while (device->run())
 	{
-		driver->beginScene(true);
+		core::smart_refctd_ptr<video::IGPUCommandBuffer> cb;
+		device->createCommandBuffers(cmdpool.get(), video::IGPUCommandBuffer::EL_PRIMARY, 1u, &cb);
+		assert(cb);
+
+		cb->begin(video::IGPUCommandBuffer::EU_ONE_TIME_SUBMIT_BIT);
+
+		asset::SViewport vp;
+		vp.minDepth = 1.f;
+		vp.maxDepth = 0.f;
+		vp.x = 0u;
+		vp.y = 0u;
+		vp.width = WIN_W;
+		vp.height = WIN_H;
+		cb->setViewport(0u, 1u, &vp);
+
+		video::IGPUQueue::SSubmitInfo info;
+		auto* cb_ = cb.get();
+		info.commandBufferCount = 1u;
+		info.commandBuffers = &cb_;
+		info.pSignalSemaphores = nullptr;
+		info.signalSemaphoreCount = 0u;
+		info.pWaitSemaphores = nullptr;
+		info.waitSemaphoreCount = 0u;
+		info.pWaitDstStageMask = nullptr;
+		queue->submit(1u, &info, nullptr);
+		cb->end();
```

Comment:
you must end() recording before submitting. Command buffer must be in executable state